tulip/abstract/discretization.py

```diff
@@ -453,6 +453,90 @@ def _plot_abstraction(ab, show_ts, only_adjacent, color_seed):
+    trans_length=1, remove_trans=False,
+    abs_tol=1e-7,
+    plotit=False, save_img=False, cont_props=None,
+    plot_every=1, simu_type='bi'
+):
+    """Refine the partition via bisimulation 
+    or dual-simulation algorithms, and establish transitions
+    based on reachability analysis. 
+
+    Reference
+    =========
+    U{[NOTM12]
+    <https://tulip-control.sourceforge.io/doc/bibliography.html#notm12>}
+
+    See Also
+    ========
+    L{prop2partition.pwa_partition}, L{prop2partition.part2convex}
+
+    @param part: L{PropPreservingPartition} object
+    @param ssys: L{LtiSysDyn} or L{PwaSysDyn} object
+    @param N: horizon length
+    @param min_cell_volume: the minimum volume of cells in the resulting
+        partition.
+    @param closed_loop: boolean indicating whether the `closed loop`
+        algorithm should be used. default True.
+    @param conservative: if true, force sequence in reachability analysis
+        to stay inside starting cell. If false, safety
+        is ensured by keeping the sequence inside a convexified
+        version of the original proposition preserving cell.
+    @param max_num_poly: maximum number of polytopes in a region to use in
+        reachability analysis.
+    @param use_all_horizon: in closed loop algorithm: if we should look
+        for reachability also in less than N steps.
+    @param trans_length: the number of polytopes allowed to cross in a
+        transition.  a value of 1 checks transitions
+        only between neighbors, a value of 2 checks
+        neighbors of neighbors and so on.
+    @param remove_trans: if True, remove found transitions between
+        non-neighbors.
+    @param abs_tol: maximum volume for an "empty" polytope
+
+    @param plotit: plot partitioning as it evolves
+    @type plotit: boolean,
+        default = False
+
+    @param save_img: save snapshots of partitioning to PDF files,
+        requires plotit=True
+    @type save_img: boolean,
+        default = False
+
+    @param cont_props: continuous propositions to plot
+    @type cont_props: list of C{Polytope}
+    
+    @param simu_type: if 'bi', use bisimulation partition; if 'dual', 
+        use dual-simulation partition
+    @type simu_type: string,
+        default = 'bi'
+
+    @rtype: L{AbstractPwa}
+    """
+    if simu_type == 'bi':
+        AbstractPwa = _discretize_bi(
+            part, ssys, N, min_cell_volume,
+            closed_loop, conservative,
+            max_num_poly, use_all_horizon,
+            trans_length, remove_trans,
+            abs_tol,
+            plotit, save_img, cont_props,
+            plot_every)
+    elif simu_type == 'dual':
+        AbstractPwa = _discretize_dual(
+            part, ssys, N, min_cell_volume,
+            closed_loop, conservative,
+            max_num_poly, use_all_horizon,
+            trans_length, remove_trans,
+            abs_tol,
+            plotit, save_img, cont_props,
+            plot_every)
```

Comment:
Conditional statements should have an `else` branch. Usually this branch raises an error, for example:

```python
else:
    raise ValueError(
        'Unknown simulation type: "{st}"'.format(
            st=simu_type))
```